cmd/kubeadm/app/cmd/phases/join/controlplaneprepare.go

```diff
@@ -220,6 +226,11 @@ func runControlPlanePrepareDownloadCertsPhaseLocal(c workflow.RunData) error {
 		return err
 	}
 
+	// If we're dry-running, download certs to tmp dir
+	if data.DryRun() {
+		cfg.CertificatesDir = data.CertificateWriteDir()
+	}
```

Comment:
same question as [1]

Reply:
Actually, it's very tricky to set CertificatesDir during the whole dry-run process. 
First, I need to download certs to the right tmp dir but I can't pass data.CertificateWriteDir() to [copycerts.DownloadCerts](https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/cmd/kubeadm/app/phases/copycerts/copycerts.go#L218) and [certsphase.CreatePKIAssets](https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/cmd/kubeadm/app/phases/certs/certs.go#L44) directly. And remake the args of those two func will be a big work.
Then, when it comes to [CreateStaticPodFiles()](https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/cmd/kubeadm/app/phases/controlplane/manifests.go#L94), CertificatesDir should be set as "/etc/kubernetes/pki" cause it will be written into static pod yaml and printed later, so I change it again